docs/guide/examples.rst

```diff
@@ -3,6 +3,48 @@ Example Archive
 
 This is a collection of code snippets that represent common GeoTrellis tasks.
 
+Get a Layer from a GeoTiff on S3
+==================================
+
+**Motivation:** You have a potentially large GeoTiff on S3 that you'd
+like to form a proper keyed GeoTrellis layer out of. While you're at it,
+you'd like to change the projection of the imagery too.
+
+New as of GeoTrellis 1.1, ``S3GeoTiffRDD`` provides a powerful set of
+functions for fetching Tiles from the cloud. Here's an example that
+encompasses your desired work-flow:
+
+.. code-block:: scala
+
+   def layerFromS3(bucket: String, key: String)(implicit sc: SparkContext): (Int, TileLayerRDD[SpatialKey]) = {
+    /* Extract Tiles efficiently from a remote GeoTiff. */
+    val ungridded: RDD[(ProjectedExtent, Tile)] =
+      S3GeoTiffRDD.spatial(bucket, key)
+
+    /* Say it was in WebMercator and we want it in ConusAlbers. */
+    val reprojected: RDD[(ProjectedExtent, Tile)] =
+      ungridded.reproject(ConusAlbers)
+
+    /* Our layer will need metadata. Luckily, this can be derived mostly for free. */
+    val (zoom, meta): (Int, TileLayerMetadata[SpatialKey]) =
+      TileLayerMetadata.fromRdd(reprojected, ZoomedLayoutScheme(ConusAlbers))
```

Comment:
I am not certain this line will actually compile, happy to be proven wrong. IIRC when using `fromRdd` functions you need to provide all the type parameters `K, V, K2`. This is the primary reason to use `.collectMetadata[SpatialKey]` form that infers the params from `reprojected` and just requires target key type, `K2`.

Reply:
This does compile, I wrote the example first in an actual editor.

Reply:
https://github.com/fosskers/playground/blob/master/scala/geotrellis/src/main/scala/playground/S3Geotiff.scala

Reply:
I withdraw my baseless accusation!